### ISSUE TITLE
npt/asphere doesn't run on specific groups

### ISSUE THREAD

**Issue by AditiKhot** [lammps/lammps#1505]
Hi,

I am trying to simulate a molecule made of one ellipsoid particle and one point particle. I assign these particles groups anisotropic and isotropic and run NPT simulation using 'fix npt/asphere' and 'fix npt' respectively. But I get the following error:
Compute temp/asphere requires extended particles (../compute_temp_asphere.cpp:112)

If I run the same simulation in NVT ensemble with 'fix nvt/asphere' and 'fix nvt', it runs perfectly.
I have attached the files. I appreciate your help!
Thank you!
[npt_asphere_issue.zip](https://github.com/lammps/lammps/files/3274588/npt_asphere_issue.zip)

**Comment by akohlmey** [comment]
Thanks for your report. The problem is, that `compute temp/asphere` currently cannot handle mixed systems of extended and point particles. It assumes, that all particles have the same numbers of degrees of freedom. This is not a problem for `fix nvt/asphere`, since it only needs to compute the temperature for the atoms in the fix group. However, for `fix npt/asphere`, it additionally needs to run `compute temp/asphere` across **all** atoms in order to correctly compute the pressure, which is a global property. 
To make this work will require some programming.

**Comment by AditiKhot** [comment]
Can you please provide some advice as to which files I should change? Or how I go about this?

Thank you!

**Comment by akohlmey** [comment]
> Can you please provide some advice as to which files I should change? Or how I go about this?

the changes would have to be in `src/ASPHERE/compute_temp_asphere.cpp` and  `src/ASPHERE/compute_temp_asphere.h`. The changes are quite involved. You need to explicitly count the number of particles that are point particles and those that are not, communicate this via MPI to all MPI ranks, then use that information to determine the degrees of freedom while correctly apply biasing and potential removal of degrees of freedom from that. On top of everything, it would have to be compatible with dynamic groups, i.e. the numbers might change at any time step.


> 
> Thank you!

**Comment by AditiKhot** [comment]
Hi, 

I had another question. Can this problem be circumvented by treating the point particle as ellipsoid with equal diameters (sphere)? Please let me know.
Also I was checking if the forces would be same in that case. So I ran two simulations:
1. One ellipsoid (type 1) and one point particle (type 2). The pairs 1-1, 1-2, and 2-2 have GayBerne, GayBerne, and LJ interactions respectively. 
The parameters are as below
pair_coeff      1       1       gayberne        0.5     3.5     1.0     3.0     1.0     1.0     3.0     1.0    
pair_coeff      1       2       gayberne        1.0     4.1     0.5     1.5     0.5     1.19    1.19    1.19   
pair_coeff      2       2       lj/cut          1.19    4.7
The coordinates and force  at step 0 are as below
ITEM: ATOMS id type x y z fx fy fz 
1 1 -4.41787 -2.47487 -2.47487 -2.40087 0.403923 0                                                                                                                                                                 
2 2 -0.531874 -2.47487 -2.47487 2.40087 -1.91755 0 
3 1 -4.41787 2.47487 -2.47487 -2.40087 -0.403923 0
4 2 -0.531874 2.47487 -2.47487 2.40087 1.91755 0 

2. Both particles are ellipsoids. The asymmetric ellipsoid particle (type 1) is same as in the simulation above and the point particle (type 2) is treated as a symmetric ellipsoid (sphere). All the pairs 1-1, 1-2, and 2-2 have GayBerne interactions. 
The parameters are as below
pair_coeff      1       1       gayberne        0.5     3.5     1.0     3.0     1.0     1.0     3.0     1.0     
pair_coeff      1       2       gayberne        1.0     4.1     0.5     1.5     0.5     1.19    1.19    1.19    
pair_coeff      2       2       gayberne        1.19    4.7     1.0     1.0     1.0     1.0     1.0     1.0     
The coordinates and forces at step 0 are as below
ITEM: ATOMS id type x y z fx fy fz                                                                                     
1 1 -4.41787 -2.47487 -2.47487 -2.24912 0.606799 0                                                                                                                                                           
2 2 -0.531874 -2.47487 -2.47487 2.24912 -1.71468 0                                                                                                                                                                       
3 1 -4.41787 2.47487 -2.47487 -2.24912 -0.606799 0 
4 2 -0.531874 2.47487 -2.47487 2.24912 1.71468 0 

However, for the same configuration, the forces are different in the two simulations.
Please let me know what might be wrong. Thank you!

**Comment by akohlmey** [comment]
yes, if all particles are ellipsoids, then there will be no error.
i cannot comment on the second part. i have no experience with gay-berne potentials.

**Comment by akohlmey** [comment]
This has been idle for too long.
